main.rb

```diff
@@ -0,0 +1,120 @@
+# rubocop: disable Style/CaseEquality
+module Enumerable
+  UNDEFINED = Object.new
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a
+    x = 0
+    while x < arr.length
+      yield(arr[x])
+      x += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = to_a
+    i = 0
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |x| result << x if yield(x) }
+    result
+  end
+
+  def my_all?(arg = UNDEFINED)
+    unless block_given?
+      if arg != UNDEFINED
+        my_each { |x| return false unless arg === x }
+      else
+        my_each { |x| return false unless x }
+      end
+      return true
+    end
+    my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  def my_any?(arg = UNDEFINED, &block)
+    unless block_given?
+      if arg != UNDEFINED
+        my_each { |x| return true if arg === x }
+      else
+        my_each { |x| return true if x }
+      end
+      return false
+    end
+    my_each { |i| return true if block.call(i) }
+    false
+  end
+
+  def my_none?(arg = UNDEFINED, &block)
+    !my_any?(arg, &block)
+  end
+
+  def my_count(arg = UNDEFINED)
+    arr = if arg.is_a? String
+            split('')
+          else
+            arg
+          end
+    count = 0
+    unless block_given?
+      if arg != UNDEFINED
+        arr.my_each { |x| count += 1 if x == arg }
+        return count
+      end
+      return length
+    end
+    my_each { |i| count += 1 if yield(i) }
+    count
+  end
```

Comment:
## my_count

- [ ] returns the number of items in enum through enumeration::range

```Ruby
range.my_count
```

_got NameError:
       undefined local variable or method `length' for 5..50:Range_

- [ ] counts the number of items in enum that are equal to the item if an argument is given

```Ruby
array.my_count(LOWEST_VALUE)
```

_got NoMethodError:
       undefined method `my_each' for 0:Integer_